### PR TITLE
Added code to preserve comments while changing name of columns

### PR DESCRIPTION
**Pull request by yashtodi94**
I have added the code to preserve comments while changing names of columns. Currently have added this code only for MySQL adapter.

I have changed the test cases accordingly.

### CODE REVIEW

**Comment by lunetics** [comment]
+1 this is important when using doctrine

**Comment by dereuromark** [comment]
Looks like a very good change. Thank you for you valuable input and improvement here.

If you resolve the conflicts we can merge this then right away IMO.

File: src/Phinx/Db/Adapter/MysqlAdapter.php [hunk 1]
```diff
@@ -414,15 +414,19 @@ public function addColumn(Table $table, Column $column)
      */
     public function renameColumn($tableName, $columnName, $newColumnName)
     {
-        $rows = $this->fetchAll(sprintf('DESCRIBE %s', $this->quoteTableName($tableName)));
+
+        $this->startCommandTimer();
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/MysqlAdapter.php]: Why do you need the timer?

Comment (yashtodi94) [reply src/Phinx/Db/Adapter/MysqlAdapter.php]: Timer wasn't added by me...I think I need to update my branch with latest master. 

Tried resolving conflicts on github itself but I don't think thats helping.

File: tests/Phinx/Db/Adapter/MysqlAdapterUnitTest.php [hunk 2]
```diff
@@ -789,10 +791,11 @@ public function testRenameColumnExists()
                      ->method('fetch')
                      ->will($this->returnValue(null));
 
-        $this->assertQuerySql("DESCRIBE `table_name`", $this->result);
+        $this->assertQuerySql("SHOW FULL COLUMNS FROM `table_name`", $this->result);
 
-
-        $this->assertExecuteSql('ALTER TABLE `table_name` CHANGE COLUMN `column_old` `column_new` int(15) NOT NULL AUTO_INCREMENT');
+        $this->assertExecuteSql(
+            'ALTER TABLE `table_name` CHANGE COLUMN `column_old` `column_new` int(15) NOT NULL AUTO_INCREMENT COMMENT \'Old Column\''
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply tests/Phinx/Db/Adapter/MysqlAdapterUnitTest.php]: You can also concat using ``'...' . '...'`

[the remark this answers is not in the document]
Comment (yashtodi94) [reply tests/Phinx/Db/Adapter/MysqlAdapterUnitTest.php]: Thanks :)

**Comment by yashtodi94** [comment]
@dereuromark Kindly review the function `getConnection()` in AdapterWrapper file. The build fails due to that function.

**Comment by dereuromark** [comment]
the fail is in master and unrelated. 
your chsnged are good 👍

**Comment by dereuromark** [comment]
Please check this one failing test case.

**Comment by yashtodi94** [comment]
@dereuromark Kindly review the latest changes. Have updated my fork with latest master and resolved all conflicts.

**Comment by lorenzo** [comment]
@yashtodi94, it looks great! Can you add a test for it too?

File: tests/Phinx/Db/Adapter/MysqlAdapterTest.php [hunk 3]
```diff
@@ -519,6 +519,25 @@ public function testRenameColumn()
         $this->assertTrue($this->adapter->hasColumn('t', 'column2'));
     }
 
+    public function testRenameColumnPreserveComment()
+    {
+        $table = new \Phinx\Db\Table('t', [], $this->adapter);
```
Comment (dereuromark) [inline tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: Ideally, the FQCN is inside the use statement here.

Comment (dereuromark) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: `$table = new Table('t', [], $this->adapter);`

Comment (yashtodi94) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: All other tests have it in `$table = new \Phinx\Db\Table('t', [], $this->adapter);` format. Should I change this one?

Comment (lorenzo) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: It’s ok, no need to change the current style right now.

**Comment by lorenzo** [comment]
Thanks!

File: tests/Phinx/Db/Adapter/MysqlAdapterTest.php (hunk 3 above)
Comment (dereuromark) [reply tests/Phinx/Db/Adapter/MysqlAdapterTest.php]: They should all be fixed at some point.
[This tool](https://github.com/spryker/code-sniffer) can help doing at :)

**Pull request merged**